keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -0,0 +1,406 @@
+
+3. **Essential data only**
+    
+    We will prioritize inclusion of only essential diagnostic and monitoring data that aligns with the intended purpose of the z-page.
+
+4. **Premature Dependency on Unstable Format**
+
+    The alpha release will explicitly support plain text format only, making it clear that the output is not intended for parsing or automated monitoring. The feature will be secured behind a feature gate that is disabled by default, ensuring users opt-in consciously. Also, the endpoint will be implemented with versioned URLs to facilitate future transitions to structured formats without breaking existing integrations, once the usage patterns and requirements are better understood
+
+## Design Details
+
+### Data Format and versioning
+
+Initially, the statusz page will exclusively support a plain text format for responses. We will implement these endpoints using versioned URLs, like /statusz/v1, to ensure future compatibility. This versioning strategy allows us to seamlessly introduce structured data formats (e.g., JSON) in the future, through a distinct endpoint such as /statusz/v2, without disrupting existing implementations.
+
+### Authz and authn
+
+Access to the endpoint will be limited to members of the `system:monitoring group` ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), which is how paths like /healthz, /livez, /readyz are restricted today. 
+
+### Endpoint Response Format
+
+#### Data format : text
+
+#### Request
+* Method: **GET** 
+* Endpoint: **/statusz/v1**
+* Body: empty
+
+#### Response fields
+* **Start Time**: The exact date and time when the component process was initiated
+* **Uptime**: The duration for which the component has been running continuously
+* **Go Version**: The specific version of the Go programming language used to build the component's binary
+* **Binary Version**: The version of the component's binary, helpful for compatibility checks and troubleshooting
+* **Emulation Version**: The Kubernetes API version which this component is emulating as
+* **Minimum Compatibility Version**: The minimum Kubernetes API version with which the component is designed to work seamlessly
+* **Useful Links**: Relative URLs to other essential endpoints, such as /healthz, /livez, /readyz, and /metrics, for deeper insights into the component's health, readiness, and performance metrics
+
+#### Sample response
+
+```
+Started: Fri Sep  6 06:19:51 UTC 2024
+Up: 0 hr 00 min 30 sec
+Go version: go1.23.0
+Binary version: 1.31.0-beta.0.981&#43;c6be932655a03b-dirty
+Emulation version: 1.31.0-beta.0.981
+Minimum Compatibility version: 1.30.0
+
+List of useful endpoints
+--------------
+healthz:/healthz
+livez:/livez
+metrics:/metrics
+readyz:/readyz
+sli metrics:/metrics/slis
+```
```

Comment:
configz?

Reply:
I dont have a plan for introducing /configz in the upcoming release, so haven't included that in here, yet.

Reply:
it is already implemented: https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/#viewing-the-kubelet-configuration

Reply:
Isn't that only for kubelet though?

Reply:
yes, kubelet only. Would be good to show this endpoint for kubelet, while not showing for other components

Reply:
Sg, added.